crates/subspace-networking/src/node.rs

```diff
@@ -272,6 +273,10 @@ impl Node {
         self.shared.id
     }
 
+    pub fn dsn_connection_observer(&self) -> watch::Receiver<bool> {
+        self.shared.dsn_connection_observer_rx.clone()
```

Comment:
Why does it need to be cloned here? Most usable methods don't require owned/mutable value.

Reply:
Return `&watch::Receiver<bool>`. If caller needs an owned value, they can call `.clone()` themselves.